### ISSUE TITLE
🐛 BUG: Image imports work in development but not in production

### ISSUE THREAD

**Issue by matiaskorhonen** [snowpackjs/astro#772]
### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

Image imports work in development but not in production.

I have a component, `Header` (`src/components/Header.astro`) that imports an image from within `src/`:

```
---
import logo from "./logo.svg";
---
<header>
    <div>
        <img width="1280" height="300" src="{logo}" alt="Logo">
    </div>
</header>
```

This seems to work just fine in development mode but `astro build` doesn't copy the asset to the `dist/` folder.

If `buildOptions.site` is set, the image doesn't get copied but if it's commented out it does.

### Steps to Reproduce

1. `npm init astro` using template ‘Starter Kit (Generic)’ with Svelte
2. Add a component that imports an image from within `src/`
3. Set `buildOptions.site` to any hostname
4. Build the site
5. Check the `dist/` folder
6. The image isn't present!

### Link to Minimal Reproducible Example (Optional)

<https://github.com/matiaskorhonen/astro-image-bug>

**Comment by woodbrettm** [comment]
I'm finding the same issue as well, even with a manual set-up.

I have certain assets that I want to run through the build, for example SVGs that I want optimized using SVGO. Those are put in an `src/assets` folder.

When running the `build` command, it looks like the markup is correctly referencing `/_astro/src/assets...`, however the assets aren't actually copied over.

The only way I'm finding for the assets to be copied is to put them in the public folder. However then I can't put them through the build process. And it seems they're not always properly loaded in the dev server when sourced from the public folder :(

**Comment by jasikpark** [comment]
This definitely feels connected to #492 to a certain degree - since if we had a set way to consumed and output compiled assets this might work easier.. though I imagine that an image included in `/src` should be copied over to the `dist/` if it is referenced without being transformed?

**Comment by woodbrettm** [comment]
I think by default that would make sense, and then layering any plugins on top of that for transformations.

**Comment by matthewp** [comment]
I think this is likely something we are getting for free via Snowpack but not handling during the build.

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
